### PR TITLE
Improve performance for align offset harness

### PR DESCRIPTION
**Pull request by carolynzech**
Currently, this harness takes 21 minutes in CI. The only point of this harness is to verify the contract for pointee types with a non-power of two byte size--17 was an arbitrary choice. Reducing to 5 and changing the solver reduces verification time to 57 seconds on my local machine.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: library/core/src/ptr/mod.rs
Comment (feliperodri) [inline library/core/src/ptr/mod.rs:2557]: Could you add a comment explain why 5 and our experience with the verification time blow up before we merge?

**Review by feliperodri — CHANGES_REQUESTED** [review]

**Review by celinval — APPROVED** [review]

File: library/core/src/ptr/mod.rs
Comment (tautschnig) [inline library/core/src/ptr/mod.rs:2559]: Can I please ask for the following additional data points to be provided?
1. You are making two changes at once. Can you instead make each change in isolation and provide the runtimes for those?
2. I am a bit surprised there is a big difference between Kani's default of cadical vs kissat. The above request will already help understand a bit better how much of a difference there really is. As we are playing with solvers, can you please also try minisat instead?

**Review by tautschnig — CHANGES_REQUESTED** [review]

**Review by feliperodri — APPROVED** [review]
LGTM modulo @tautschnig comments.

File: library/core/src/ptr/mod.rs
Comment (carolynzech) [reply library/core/src/ptr/mod.rs:2559]: | Stride | Solver | Runtime (sec) |
|-------|--------|-------------|
| 5 | cadical | 205.23
| 5| kissat |  58.33 |
| 5 | minisat | 528.62 |
| 17 | cadical | 1126.87 |
| 17 | kissat | 495.80 |
| 17| minisat | cut off after 2.5 hours |

Comment (carolynzech) [reply library/core/src/ptr/mod.rs:2559]: I still have minisat running in the background, so I'll come back and update this if it finishes by the end of the day.

Comment (carolynzech) [reply library/core/src/ptr/mod.rs:2559]: Decided to cut it off after 4hrs -- this is enough to get the point that minisat doesn't do well for this harness.

**Pull request merged**
